torchmetrics/metric.py

```diff
@@ -305,10 +310,10 @@ def sync_context(
             for attr, val in cache.items():
                 setattr(self, attr, val)
 
-    def _wrap_compute(self, compute):
+    def _wrap_compute(self, compute: Callable) -> Callable:
 
         @functools.wraps(compute)
-        def wrapped_func(*args, **kwargs):
+        def wrapped_func(*args: Any, **kwargs: Any) -> Any:
```

Comment:
```suggestion
        def wrapped_func() -> Any:
```

Can also match `compute` signature